problems/loops.js

```diff
@@ -82,59 +140,93 @@ function tillXTimes9WhileLoop() {}
 * 4 * 9 = 36
 * 5 * 9 = 45
  */
-function tillXTimes9ForLoop() {}
-
+function tillXTimes9ForLoop(x) {
+  let newArray = [];
+  for (let num = 0; num <= x; num++) {
+    newArray.push(num * 9);
+  }
+  return newArray;
+}
 
 /**
  * Returns an array with all the numbers that end in 5 from 1 to 100
 * Must be done with a while loop
  * @returns {number[]} [5, 15, 25, ..., 95]
 
  */
-function endsWithFiveWhileLoop() {}
-
-
+function endsWithFiveWhileLoop() {
+  let num = 0;
+  let newArray = [];
+  while (num <= 100) {
+    if (num % 10 === 5) {
+      newArray.push(num);
+    }
+    num++;
+  }
+  return newArray;
+}
 
 /**
  * Returns an array with all the numbers that end in 5 from 1 to 100
 * Must be done with a for loop
  * @returns {number[]} [5, 15, 25, ..., 95]
 
  */
-function endsWithFiveForLoop() {}
-// Try refactoring the code above to increment your i by 10 and starting at 5.
-
+function endsWithFiveForLoop() {
+  let newArray = [];
+  for (let num = 0; num <= 100; num++) {
+    if (num % 10 === 5) {
+      newArray.push(num);
+    }
+  }
+  return newArray;
+}
 
+// Try refactoring the code above to increment your i by 10 and starting at 5.
 
-/**
+/** (let num = 5; num <= 100; num + 10)
  Without running/executing your code, how many times will the loop below run?
   Explain why.
- */
+ /** it will be ran 9 times because once you increment by 10 you can go into 100 only 9 times. Once you hit 95 you can no longer
+  * increment by 10
 /*
 let i = 5;
-
 while (i > 3) {
     i += 1
 }
 */
 
 /**
  * // Write your explanation here!
- *
+ * This will create an infinite loop because you have set no max value. I will continue to be incremented because the condition
+ * says while i is bigger than 3, and we already set i's value to 5 when we started the loop
  */
 
- /**
+/**
  * Returns an array with all the numbers from all the numbers from 1 to 100.
  * For numbers divisible by 3, add "Fizz" instead of the number into the array.
  * For numbers divisible by 5 (and not 3), add "Buzz" instead of the number into the array.
 
  * @returns {Array} [1, 2, "Fizz", 4, "Buzz"... 98, "Fizz", "Buzz"]
 
  */
-function fizzBuzzPart1() {}
-
+function fizzBuzzPart1() {
+  let newArray = [];
+  let num = 1;
+  while (num <= 100) {
+    if (num % 3 === 0) {
+      newArray.push("Fizz");
+    } else if (num % 5 === 0 && num % 3 != 0) {
```

Comment:
Don't need the right side of the && if it was divisible by 3 we would have gone into the  first if.